packages/preact-iso/router.js

```diff
@@ -82,20 +104,28 @@ export function Router(props) {
 		} else commit();
 	}, [url]);
 
-	const children = [].concat(...props.children);
-
-	let a = children.filter(c => c.props.path === path);
-
-	if (a.length == 0) a = children.filter(c => c.props.default);
-
-	curChildren.current = a.map((p, i) => cloneElement(p, { path, query }));
+	let p, d, m;
+	[].concat(props.children || []).some(vnode => {
+		const matches = exec(path, vnode.props.path, m = { path, query });
+		if (matches) {
+			return p = h(
+				RouteContext.Provider,
+				{ value: {  ...matches } },
+				cloneElement(vnode, { ...m, ...matches })
```

Comment:
`m` and `matches` are the same object here. also if we can avoid using object spread, that preserves Safari 11 and Edge support. Since `cloneElement()` clones props anyway, it should be safe to skip manual object cloning here altogether:

```suggestion
				{ value: m },
				cloneElement(vnode, m)
```